RealmSwift/SwiftUI.swift

```diff
@@ -847,38 +917,36 @@ public enum AsyncOpenState {
      - parameter partitionValue: The `BSON` value the Realm is partitioned on.
      - parameter configuration: The `Realm.Configuration` used when creating the Realm,
                  user's sync configuration for the given partition value will be set as the `syncConfiguration`,
-                 if empty the configuration is set to the `defaultConfiguration`
+                 if empty the user configuration will be used.
```

Comment:
Isn't this a breaking change? If we previously were using settings from the default configuration and now aren't anyone who was relying on that will have problems.

Reply:
This sounds like a bug in the original implementation– I believe we were always supposed to use the user's configuration.

Reply:
yep as mentioned on the PR's description, this was a bug on the original implementation
`We are not setting defaultConfiguration as default if the user doesn't specified one in the initialisation of the property wrapper, this was causing the fileURL to be the one provided by the defaultConfiguration and never the one from the user configuration. This should only be valid if the user provides a configuration via the constructor, overriding the user configuration path.`